packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryBuffer.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+// Default duration threshold for reporting performance entries (0 means "report
+// all")
+constexpr double DEFAULT_DURATION_THRESHOLD = 0.0;
+
+/**
+ * Abstract performance entry buffer with reporting flags.
+ * Subtypes differ on how entries are stored.
+ */
+class PerformanceEntryBuffer {
+public:
+  double durationThreshold{DEFAULT_DURATION_THRESHOLD};
+  size_t droppedEntriesCount{0};
+
+  explicit PerformanceEntryBuffer() = default;
+  virtual ~PerformanceEntryBuffer() = default;
+
+  virtual void add(const PerformanceEntry& entry) = 0;
+  virtual void consume(std::vector<PerformanceEntry>& target) = 0;
```

Comment:
Nit: if this is essentially the same as `getEntries(std::nullopt, target)` we could as well name it the same way. Just having 2 methods (and define one inline):

```
void getEntries(std::vector<PerformanceEntry>& target) {
  return getEntries(std::nullopt, target);
}
```

Reply:
Wait, I see that in the implementation of this method in the different subclasses, you actually clear the contents of the buffer. What's the use case for that? AFAIK these buffers shouldn't be cleared, unless you specifically call `clear`.